pyVMC.py

```diff
@@ -5796,14 +5798,21 @@ def __init__(self,prog):
         pass
 
     # Depending on params in config.ini use OAuth or Refresh Token for auth
-    params = vars(args)
-    if len(clientId) != 0:
-        sessiontoken = getAccessToken(oauth_clientId=clientId,oauth_clientSecret=clientSecret)
-    else:
-        sessiontoken = getAccessToken(myKey=Refresh_Token)
+    auth_method = args.auth_method
+    sessiontoken = ''
+    match auth_method:
+        case "oauth":
+            auth_params = {'auth_method':auth_method, 'strCSPProdURL':strCSPProdURL, 'oauth_clientSecret':clientSecret, 'oauth_clientId':clientId}
+            sessiontoken = getAccessToken(**auth_params)
+        case "refresh_token":
+            auth_params = {'auth_method':auth_method, 'strCSPProdURL':strCSPProdURL, 'myKey':Refresh_Token}
+            sessiontoken = getAccessToken(**auth_params)
```

Comment:
I think the placement here means that command-line overrides will be ignored.

Reply:
Can you expand?

Reply:
I think by physically placing this block on these line numbers, above the section of code where the command-line overrides are defined, the command-line overrides will be ignored. Meaning the params.update statements that are reading command-line override options, just  below this block of code.

Reply:
Ah - gotcha.  I don't think this is an issue...  I built it this way because the only thing that needs to go into the params dictionary is the sessiontoken itself.  getAccessToken needs to run before the params dictionary is built - you will note I use a separate dictionary - 'auth_params' - for calling getAccessToken, and it still uses the values from config.ini

Reply:
How would someone pass those at the command line?

Reply:
I guess my point is, there are no command-line overrides for those variables.  They are ONLY pulled in from config.ini.  So unless someone writes the code to do so, how would they provide some kind of override?  What am I missing?

Reply:
@fr0gger03 has a point.. there is no command line option for for the URLs at the moment. But I agree that if they are passed in they should take priority over the config.ini version.

Reply:
OK - check out the latest push to this PR... 

1. made the OAuth optional, with refresh_token as default
2. in the event the arguments for oauth / refresh_token haven't been added to subcommands, default to refresh_token
3. changed the names on the global flags to remove confusion